### PR TITLE
Adaptors_task_Gershkul

### PR DESCRIPTION
**Pull request by DevASG**
Implementation

### CODE REVIEW

File: adaptors_task/student/src/impl.cpp [hunk 1]
```diff
@@ -1,8 +1,35 @@
...
+        if (it == brackets.end()) {
+            continue;
+        }
+        if (it->first == bracket) {
+            stack.push(bracket);
+        } else {
+            if (stack.empty()) {
+                return false;
+            }
+            if (stack.top() != it->first) {
```
Comment (smay1613) [inline adaptors_task/student/src/impl.cpp:28]: please combine this condition with the above

File: adaptors_task/student/src/impl.cpp [hunk 2]
```diff
@@ -1,8 +1,35 @@
...
+{
+    std::stack<char> stack;
+    for (const auto& bracket : source) {
+        auto it = std::find_if(brackets.begin(), brackets.end(), [&](std::pair<char, char> const p) {
+            return p.first == bracket || p.second == bracket;
+        });
+        if (it == brackets.end()) {
+            continue;
+        }
+        if (it->first == bracket) {
```
Comment (smay1613) [inline adaptors_task/student/src/impl.cpp:22]: please name this condition

**Review by smay1613 — CHANGES_REQUESTED** [review]

File: adaptors_task/student/src/impl.cpp (hunk 1 above)
Comment (DevASG) [reply adaptors_task/student/src/impl.cpp]: Done

File: adaptors_task/student/src/impl.cpp (hunk 2 above)
Comment (DevASG) [reply adaptors_task/student/src/impl.cpp]: Done

**Comment by smay1613** [comment]
Nice!

**Pull request closed without merging**